lib/jsonapi/utils/underscore.ex

```diff
@@ -56,14 +60,44 @@ defmodule JSONAPI.Utils.Underscore do
   end
 
   def underscore(value) when is_map(value) do
-    Enum.into(value, %{}, &underscore/1)
+    Enum.into(value, %{}, &underscore(&1))
   end
 
   def underscore({key, value}) do
-    if is_map(value) do
-      {underscore(key), underscore(value)}
+    if underscore?(key) do
```

Comment:
Could this be simplified in such a way to remove the need for a nested `if`?

```elixir
if underscore?(key) && is_map(value) do
  {underscore(key), underscore(value)}
else
  {underscore(key), value}
end	   
```

Reply:
yes!